Show original and changed linefeed in diff view

Chris2011 opened the pull request:
The current behavior, when the linefeed changes, you just see, that the file was modified but you dunno why. This PR adds the original and the changed linefeed.

I used GridLayout for the header because I couldn't handle adding the exact position what it is now. But this changes the layout a bit, You can see it in the 


---
**^Add meaningful description above**

<details open>
<summary>Click to collapse/expand PR instructions</summary>

If you're a committer, please label the PR before pressing "Create pull request" so that the right test jobs can run.

### PR approval and merge checklist:

1. [ ] Was this PR [correctly labeled](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=240884239#PRsandYouAreviewerGuide-PRtriggeredCIJobs(conditionalCIpipeline)), did the right tests run? When did they run?
2. [ ] Is this PR [squashed](https://cwiki.apache.org/confluence/display/NETBEANS/git%3A+squash+and+merge)?
3. [ ] Are author name / email address correct? Are [co-authors](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors#creating-co-authored-commits-on-the-command-line) correctly listed? Do the commit messages need updates?
3. [ ] Does the PR title and description still fit after the Nth iteration? Is the description sufficient to appear in the release notes?

If this PR targets the delivery branch: [don't merge](https://cwiki.apache.org/confluence/display/NETBEANS/Pull+requests+for+delivery). ([full wiki article](https://cwiki.apache.org/confluence/display/NETBEANS/PRs+and+You+-+A+reviewer+Guide))

</details>

Chris2011 commented on ide/diff/src/org/netbeans/modules/diff/builtin/visualizer/editable/EditableDiffView.java:
```diff
@@ -192,6 +198,9 @@ public EditableDiffView(final StreamSource ss1, final StreamSource ss2, boolean
         actionsEnabled = ss2.isEditable();
         diffMarkprovider = new EditableDiffMarkProvider();        
 
+        lineEnding1 = detectLineEnding(ss1);
+        lineEnding2 = detectLineEnding(ss2);
```
Thx for the hint, will have a lool soon :)

Chris2011 replied:
Updated and it seems good enough.

matthiasblaesing commented on ide/diff/src/org/netbeans/modules/diff/builtin/visualizer/editable/EditableDiffView.java:
```diff
@@ -372,6 +392,38 @@ public void run () {
         }
     }
 
+    private static String detectLineEnding(StreamSource source) {
+        try {
+            FileObject fo = source.getLookup().lookup(FileObject.class);
+            DataObject dao = DataObject.find(fo);
+
+            EditorCookie editorCookie = dao.getCookie(EditorCookie.class);
+            if (editorCookie == null) {
+                return null;
+            }
+
+            Document doc;
+            try {
+                doc = editorCookie.openDocument();
+            } catch (IOException ex) {
+                return null;
+            }
```
This will fail if the source is not backed by a file. If you pass in the `DiffContentPanel`, you can simplify this to:


```suggestion
    private static String detectLineEnding(DiffContentPanel source) {
        try {
            Document doc = source.getEditorPane().getDocument();
```

The panels are directly available in `RefreshDiffTask#diff`.

matthiasblaesing commented on ide/diff/src/org/netbeans/modules/diff/builtin/visualizer/editable/EditableDiffView.java:
```diff
@@ -177,10 +190,17 @@ public EditableDiffView (final StreamSource ss1, final StreamSource ss2) {
-        if (title1 == null) title1 = NbBundle.getMessage(EditableDiffView.class, "CTL_DiffPanel_NoTitle"); // NOI18N
-        String title2 = ss2.getTitle();
-        if (title2 == null) title2 = NbBundle.getMessage(EditableDiffView.class, "CTL_DiffPanel_NoTitle"); // NOI18N
+
+        source1 = ss1;
+        source2 = ss2;
```
If the `DiffContentPanel`s are used, this can be removed.

matthiasblaesing commented on ide/diff/src/org/netbeans/modules/diff/builtin/visualizer/editable/EditableDiffView.java:
```diff
@@ -1518,6 +1587,20 @@ public void run() {
+                            lineEndingLabel1.setText("<html><strong style='background-color: " + String.format("#%02x%02x%02x", colorChanged.getRed(), colorChanged.getGreen(), colorChanged.getBlue()) + "'>" + lineEnding1 + "</strong></html>");
+                            lineEndingLabel2.setText("<html><strong style='background-color: " + String.format("#%02x%02x%02x", colorChanged.getRed(), colorChanged.getGreen(), colorChanged.getBlue()) + "'>" + lineEnding2 + "</strong></html>");
+                        }
+
+                        setSourceTitle(fileLabel1, title1);
+                        setSourceTitle(fileLabel2, title2);
```
This is exactly doing, what was already done in the constructor. Thus this can be removed.

matthiasblaesing commented on ide/diff/src/org/netbeans/modules/diff/builtin/visualizer/editable/EditableDiffView.java:
```diff
@@ -166,6 +173,12 @@ public class EditableDiffView extends DiffControllerImpl implements DiffView, Do
     private final String name1;
     private final String name2;
+
+    private final StreamSource source1;
+    private final StreamSource source2;
+    private final String title1;
+    private final String title2;
+
```
These are not required (see comments below).

matthiasblaesing commented on ide/diff/src/org/netbeans/modules/diff/builtin/visualizer/editable/EditableDiffView.java:
```diff
@@ -122,13 +124,18 @@ public class EditableDiffView extends DiffControllerImpl implements DiffView, Do
     final JComponent view;
     final JSplitPane jSplitPane1 = new JSplitPane();
 
+    private String lineEnding1;
+    private String lineEnding2;
+
```
These are not required. `lineEnding1` and `lineEnding2` are only required in `RefreshDiffTask#run` and thus can be declared there.

matthiasblaesing reviewed (commented):
Nice feature and works.

Independent of this change NetBeans will not detect changes to the line endings if the document is already opened (`BaseDocument` initializes the property exactly once). This is a different problem though.

For the implementation I left a few inline comments.

Chris2011 commented:
Everything changed and tested as requested. Thx.

Chris2011 commented:
I already added a background to make it more prominent. I already wanted to use the colors from the git integration for added, deleted and modified as forground colors for just the text, but they didn't fit on the light theme.

matthiasblaesing reviewed (approved):
Looks sane to me. Thank you. Please squash before merging.
